src/sentry/http.py

```diff
@@ -108,25 +108,49 @@ def send(self, request, *args, **kwargs):
         return super(BlacklistAdapter, self).send(request, *args, **kwargs)
 
 
-def build_session():
-    session = requests.Session()
-    session.headers.update({'User-Agent': USER_AGENT})
-    session.mount('https://', BlacklistAdapter())
-    session.mount('http://', BlacklistAdapter())
-    return session
+class SafeSession(requests.Session):
+
+    def __init__(self):
+        requests.Session.__init__(self)
+        self.headers.update({'User-Agent': USER_AGENT})
+        self.mount('https://', BlacklistAdapter())
+        self.mount('http://', BlacklistAdapter())
+
+    def request(self, *args, **kwargs):
+        kwargs.setdefault('timeout', 30)
+        try:
+            response = requests.Session.request(self, *args, **kwargs)
+        # Our version of requests does not transform ZeroReturnError into an
+        # appropriately generically catchable exception
+        except ZeroReturnError as exc:
+            import sys
+            exc_tb = sys.exc_info()[2]
+            six.reraise(SSLError, exc, exc_tb)
+            del exc_tb
+        # requests' attempts to use chardet internally when no encoding is found
+        # and we want to avoid that slow behavior
+        if not response.encoding:
+            response.encoding = 'utf-8'
+        return response
+
+
+build_session = SafeSession
 
 
 def safe_urlopen(url, method=None, params=None, data=None, json=None,
                  headers=None, allow_redirects=False, timeout=30,
-                 verify_ssl=True, user_agent=None):
+                 verify_ssl=True, user_agent=None, session=None):
```

Comment:
Is there a reason now to still allow this? Seems like you'd just use `SafeSession` directly if you want to reuse the Session object and `SafeSession` now contains all the logic that was repeated.